lib/fastlane/actions/add_icon_overlay.rb

```diff
@@ -0,0 +1,181 @@
+module Fastlane
+  module Actions
+
+    class AddIconOverlayAction < Action
+      def self.run(params)
+        Helper.log.info "Image to overlay on icons: #{params[:overlay_image_path]}" unless not params[:overlay_type] == 'image'
+        require 'mini_magick'
+
+        overlay_type = params[:overlay_type]
+        appiconset = params[:appiconset_path]
+
+        if overlay_type == 'image'
+          self.addImageOverlay(appiconset, params)
+        elsif overlay_type == 'banner'
+          self.addBannerOverlay(appiconset, params)
+        else
+          self.addTextOverlay(appiconset, params)
+        end
+      end
+
+      def self.addTextOverlay(appiconset, params)
+        Dir.glob(File.join(appiconset, '*.png')) do |icon|
+          img = MiniMagick::Image.new(icon)
+          MiniMagick::Tool::Convert.new do |c|
+            height = img[:height]*params[:text_overlay_height].to_f
+            c.background(params[:text_background_color])
+            c.fill(params[:text_foreground_color])
+            c.gravity('center')
+            c.size("#{img[:width]}x#{height}")
+            c.font(params[:text_overlay_font])
+            c << "caption:#{params[:text_overlay]}"
+            c << icon
+            c.swap.+
+            c.gravity(params[:text_overlay_position])
+            c.composite(icon)
+          end
+        end
+      end
+
+      def self.addBannerOverlay(appiconset, params)
+        Dir.glob(File.join(appiconset, '*.png')) do |icon|
+          original_icon = MiniMagick::Image.new(icon)
+          width = original_icon.width
+          height = original_icon.height
+          rect_height = height * params[:text_overlay_height].to_f
+          rect_width = width * 1.6
+          x = width * 0.65 # correct positioning for topright banner
+          y = -0.7 * rect_height # correct positioning for topright banner
+          text_size = width / 8
+          padding = text_size / 2
+
+          result = original_icon.combine_options do |c|
+            c.fill(params[:text_background_color])
+            c.draw "push graphic-context translate #{x}, #{y} rotate 45 rectangle #{0}, #{0}, #{0 + rect_width}, #{0 + rect_height} pop graphic-context"
+          end
+
+          result = result.combine_options do |c|
+            c.fill(params[:text_foreground_color]) # set the text fill colour
+            c.gravity 'Northeast' # anchor the text to the top right
+            c.font(params[:text_overlay_font])
+            c.draw "rotate 45 text #{text_size + padding},#{text_size + padding} '#{params[:text_overlay]}'" # draw text
+            c.pointsize "#{text_size}" # set font size
+          end
+
+          result.write icon
+        end
+      end
+
+      def self.addImageOverlay(appiconset, params)
```

Comment:
Already fixed with e0e2bb3f35cfd00ad9401eea4a7e4b72dffcb87a but `HoundCI` failed to pick the change